api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -0,0 +1,410 @@
+import org.openmrs.module.emrapi.diagnosis.Diagnosis;
+import org.openmrs.module.emrapi.diagnosis.DiagnosisMetadata;
+import org.openmrs.module.emrapi.disposition.DispositionType;
+import org.openmrs.module.emrapi.visit.VisitDomainWrapper;
+import org.openmrs.module.htmlformentry.FormEntryContext;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+import org.openmrs.module.htmlformentry.FormSubmissionActions;
+import org.openmrs.module.htmlformentry.FormSubmissionError;
+import org.openmrs.module.htmlformentry.InvalidActionException;
+import org.openmrs.module.htmlformentry.action.FormSubmissionControllerAction;
+import org.openmrs.module.htmlformentry.element.HtmlGeneratorElement;
+import org.openmrs.module.htmlformentry.widget.ErrorWidget;
+import org.openmrs.module.htmlformentry.widget.HiddenFieldWidget;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.page.PageAction;
+
+public class EncounterDiagnosesElement implements HtmlGeneratorElement, FormSubmissionControllerAction {
+
+    private boolean required = false;
+    private UiUtils uiUtils;
+    private String selectedDiagnosesTarget;
+
+    private EmrApiProperties emrApiProperties;
+
+    private ConceptService conceptService;
+
+    private AdtService adtService;
+
+    private DispositionType dispositionTypeForPriorDiagnoses = null;
+
+    // we do not actually use the hiddenDiagnoses widget (the form field name is hardcoded) but we need it to register errorWidget
+    private HiddenFieldWidget hiddenDiagnoses;
+    private ErrorWidget errorWidget;
+
+    public EncounterDiagnosesElement() {
+    }
+
+    /**
+     * Method to convert the core diagnosis object into a list of diagnoses compatible with the diagnosis object in the emrapi module
+     * @return diagnoses
+     * */
+    private List<Diagnosis> convert(List<org.openmrs.Diagnosis> coreDiagnoses) {
+        List<Diagnosis> diagnoses = new ArrayList<Diagnosis>();
+        for (org.openmrs.Diagnosis coreDiagnosis : coreDiagnoses) {
+            Diagnosis diagnosis = new Diagnosis();
+            CodedOrFreeText coded = coreDiagnosis.getDiagnosis();
+            diagnosis.setDiagnosis(new CodedOrFreeTextAnswer(coded.getCoded(), coded.getSpecificName(), coded.getNonCoded()));
+            diagnosis.setCertainty(coreDiagnosis.getCertainty() == ConditionVerificationStatus.CONFIRMED ? Diagnosis.Certainty.CONFIRMED : Diagnosis.Certainty.PRESUMED);
+            diagnosis.setOrder(coreDiagnosis.getRank() == 1 ? Diagnosis.Order.PRIMARY : Diagnosis.Order.SECONDARY);
+            diagnoses.add(diagnosis);
+        }
+        return diagnoses;
+    }
+
+    @Override
+    public String generateHtml(FormEntryContext context) {
+        List<Diagnosis> existingDiagnoses = convert(Context.getDiagnosisService().getDiagnoses(context.getExistingPatient(), null));
+        //List<Diagnosis> existingDiagnoses = getExistingDiagnoses(context, emrApiProperties.getDiagnosisMetadata());
```

Comment:
You can remove the above commented out line.

Reply:
@dkayiwa I have removed this and the other blocks of code that were commented out. I hope that's okay